daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -718,3 +741,196 @@ class UnparseResult(dp: DataProcessor, ustate: UState)
+    entityResolver = resolver
+  }
+
+  override def getEntityResolver: EntityResolver = entityResolver
+
+  override def setDTDHandler(handler: DTDHandler): Unit = {
+    dtdHandler = handler
+  }
+
+  override def getDTDHandler: DTDHandler = dtdHandler
+
+  override def setContentHandler(handler: ContentHandler): Unit = {
+    contentHandler = handler;
+  }
+
+  override def getContentHandler: ContentHandler = contentHandler
+
+  override def setErrorHandler(handler: ErrorHandler): Unit = {
+    errorHandler = handler;
+  }
+
+  override def getErrorHandler: ErrorHandler = errorHandler
+
+  override def parse(input: InputSource): Unit = {
+    val is = input.getByteStream
+    if(is != null) {
+      val isdis = InputSourceDataInputStream(is)
+      parse(isdis)
+    } else {
+      throw new IOException("Inputsource must be backed by Inputstream")
+    }
+  }
+
+  override def parse(systemId: String): Unit = {
+    throw new IOException("SAX parsing of systemId is unsupported")
+  }
+
+  def parse(isdis: InputSourceDataInputStream): Unit = {
+    val sio = createSAXInfosetOutputter(this)
+    val pr = dp.parse(isdis, sio)
+    handleDiagnostics(pr)
+    setProperty(s"${XMLUtils.DAFFODIL_SAX_URN_ROOT}:ParseResult", pr)
+  }
+
+  def parse(stream: InputStream): Unit = {
+    val isdis = InputSourceDataInputStream(stream)
+    parse(isdis)
+  }
+
+  def parse(arr: Array[Byte]): Unit = {
+    val isdis = InputSourceDataInputStream(arr)
+    parse(isdis)
+  }
+
+  private def handleDiagnostics(pr: DFDL.ParseResult): Unit = {
+    val diagnostics = pr.getDiagnostics
+    val eh = this.getErrorHandler
+    if (diagnostics.nonEmpty && eh != null) {
+      diagnostics.foreach { d =>
+        val spe = {
+          val msg = d.getMessage()
+          val (lineNo, colNo, systemId) = d.getLocationsInSchemaFiles.headOption.map { s =>
+            val sl = s.asInstanceOf[SchemaFileLocation]
+            val ln = sl.lineNumber.getOrElse("0").toInt
+            val cn = sl.columnNumber.getOrElse("0").toInt
+            val sId = sl.uriString
+            (ln, cn, sId)
+          }.getOrElse((0,0, null))
+
+          val spe = new SAXParseException(msg, null, systemId, lineNo, colNo, d)
+          spe
+        }
+
+        if (d.isError) {
+          eh.error(spe)
+        } else {
+          eh.warning(spe)
+        }
+      }
+    }
+  }
+
+  /**
+   * Creates SAXInfosetOutputter object and attempts to setBlobAttributes on it if
+   * it has at least the blobDirectory property set
+   *
+   * @return SAXInfosetOutputter object with or without blob Attributes set
+   */
+  private def createSAXInfosetOutputter(xmlReader: DaffodilXMLReader): SAXInfosetOutputter = {
+    val sioo = new SAXInfosetOutputter(xmlReader)
+    val siof = try {
+      val blobDir = try {
+        getProperty(s"${XMLUtils.DAFFODIL_SAX_URN_ROOT}:BlobDirectory").asInstanceOf[java.nio
+        .file.Path]
+      } catch {
```

Comment:
Setting the propety should have done the appropriate checks. So I don't think this should need any try/catch stuff. Just needs to use the saxProperty.. variables that youve set. So I think this can all just become

```scala
sioo.setBlobAttributes(saxPropertyBlobDir, saxPropertyBlobPrefix, saxPropetyBlobSuffix)
```
The default values and checking in setProperty always ensure we have a correct value.